File: loginprofile/app/src/main/java/com/quang436/loginprofile/MainActivity.java
package com.quang436.loginprofile;

import android.content.Intent;
import android.os.Bundle;
import android.text.TextUtils;
import android.widget.TextView;
import android.widget.Toast;

import androidx.appcompat.app.AppCompatActivity;

import com.google.android.material.button.MaterialButton;
import com.google.android.material.textfield.TextInputEditText;

public class MainActivity extends AppCompatActivity {

    private TextInputEditText edtUser, edtPass;
    private MaterialButton btnLogin;
    private TextView tvForgot;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        edtUser = findViewById(R.id.edtUser);
        edtPass = findViewById(R.id.edtPass);
        btnLogin = findViewById(R.id.btnLogin);
        tvForgot = findViewById(R.id.tvForgot);

        btnLogin.setOnClickListener(v -> doLogin());
        tvForgot.setOnClickListener(v ->
                Toast.makeText(this, "Demo: liên hệ admin để reset mật khẩu!", Toast.LENGTH_SHORT).show()
        );
    }

    private void doLogin() {
        String user = edtUser.getText() == null ? "" : edtUser.getText().toString().trim();
        String pass = edtPass.getText() == null ? "" : edtPass.getText().toString();

        if (TextUtils.isEmpty(user)) {
            edtUser.setError("Nhập User Name");
            return;
        }
        if (TextUtils.isEmpty(pass)) {
            edtPass.setError("Nhập Password");
            return;
        }

        // Demo: coi như đăng nhập thành công, chuyển sang Profile
        Intent i = new Intent(this, ProfileActivity.class);
        i.putExtra("username", user);
        i.putExtra("email", user.toLowerCase() + "@mail.com");
        i.putExtra("phone", "+84 999 555 222");
        i.putExtra("skype", user.toLowerCase());
        i.putExtra("web", "https://example.com/" + user.toLowerCase());
        startActivity(i);
    }
}

File: BT1/app/src/main/java/com/example/myapplication/MainActivity.kt
package com.example.myapplication

import androidx.appcompat.app.AppCompatActivity
import android.os.Bundle
import android.widget.*

class MainActivity : AppCompatActivity() {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)

        // Khởi tạo basic functionality
        setupBasicEvents_332()
    }

    private fun setupBasicEvents_332() {
        val loginButton_332 = findViewById<Button>(R.id.login_button_332)
        val usernameInput_332 = findViewById<EditText>(R.id.username_input_332)
        val passwordInput_332 = findViewById<EditText>(R.id.password_input_332)
        val statLeft_332 = findViewById<TextView>(R.id.stat_left_332)
        val statRight_332 = findViewById<TextView>(R.id.stat_right_332)

        // Click login button
        loginButton_332.setOnClickListener {
            val username = usernameInput_332.text.toString()
            if (username.isNotEmpty()) {
                Toast.makeText(this, "Welcome $username!", Toast.LENGTH_SHORT).show()
            } else {
                Toast.makeText(this, "Please enter username", Toast.LENGTH_SHORT).show()
            }
        }

        // Click stats
        statLeft_332.setOnClickListener {
            val current = statLeft_332.text.toString().toIntOrNull() ?: 125
            statLeft_332.text = (current + 1).toString()
        }

        statRight_332.setOnClickListener {
            val current = statRight_332.text.toString().toIntOrNull() ?: 250
            statRight_332.text = (current + 1).toString()
        }
    }
}